perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -184,7 +199,15 @@ public void setUserExtSourceAttributes(PerunSession session, UserExtSource ues,
 							attributeWithValue = attributesManagerBl.getAttribute(session, ues, attr.getName());
 						}
 					}
-					attributeWithValue.setValue(attrValue);
+
+					//for Array list attributes try to parse string value into individual fields
+					if(attributeWithValue.getType().equals(ArrayList.class.getName())) {
```

Comment:
We have also LargeArrayList type of attributes, which are technically lists, but can be larger than 4000bytes. You can use constant ``largeArrayListClassName`` from ``BeansUtils`` to check the type against it.

Reply:
Fixed.

Reply:
I probably wasn't exact. You need to check both ``ArrayList`` and ``LargeArrayList`` types and handle them as list.